src/Rpc/NodeGuardService.cs

```diff
@@ -859,4 +864,30 @@ public override async Task<GetWithdrawalsRequestStatusResponse> GetWithdrawalsRe
             }
         };
     }
+
+    public override async Task<GetChannelResponse> GetChannel(GetChannelRequest request, ServerCallContext context)
+    {
+        var channel = await _channelRepository.GetById(request.ChannelId);
+        if (channel == null)
+        {
+            throw new RpcException(new Status(StatusCode.NotFound, "Channel not found"));
+        }
+        
+        var result = new GetChannelResponse()
+        {
+            FundingTx = channel.FundingTx,
+            OutputIndex = channel.FundingTxOutputIndex,
+            ChanId = channel.ChanId,
+            SatsAmount = channel.SatsAmount,
+            Status = (CHANNEL_STATUS)((int)channel.Status - 1),
+            CreatedByNodeguard = channel.CreatedByNodeGuard,
+            IsAutomatedLiquidityEnabled = channel.IsAutomatedLiquidityEnabled,
+            IsPrivate = channel.IsPrivate,
+        };
+
+        if (channel.BtcCloseAddress != null)
+            result.BtcCloseAddress = channel.BtcCloseAddress;
```

Comment:
Maybe moving this to GetChannelResponse object creation as a ternary operator:

`BtcCloseAddress = channel.btcCloseAddress != null ? channel.BtcCloseAddress : String.Empty` 

this way we avoid having that if :)